src/client.rs

```diff
@@ -415,6 +497,41 @@ impl Client {
         }
     }
 
+    /// Return a `futures::TryStream` of results from a query.
+    ///
+    /// # Example
+    ///
+    /// You can iterate over the rows from a query using a `while` loop:
+    /// ```no_run
+    /// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
+    /// use prusto::{ Presto,Client };
+    /// use futures::stream::TryStreamExt;
+    ///
+    /// #[derive(Presto)]
+    /// struct MyRow {
+    ///   a: i32,
+    ///   b: String,
+    /// }
+    ///
+    /// let client: Client = unimplemented!();
+    ///
+    /// let results = client.stream::<MyRow>("SELECT * FROM my_table".to_string());
+    /// while let Some(MyRow { a, b }) = results.try_next().await? {
+    ///     println!("This row has {a} and {b}");
+    /// }
+    /// # Ok(())
+    /// # }
+    pub fn stream<T: Presto + 'static>(
+        &self,
+        sql: String,
+    ) -> impl futures::stream::Stream<Item = Result<T>> + std::marker::Unpin + '_ {
+        let stream = MyStream {
+            inner: QueryInternal::Query { sql, client: self },
+            phantom: std::marker::PhantomData,
+        };
+        Box::pin(futures::stream::try_unfold(stream, MyStream::next_and_self))
+    }
```

Comment:
I think we should remove `std::marker::Unpin` on the return type and let user decide which pin strategy to use, e.g. `pin!` macro,`Box::pin`.